SerialPrograms/Source/Integrations/DppIntegration/DppClient.cpp

```diff
@@ -152,11 +151,14 @@ void Client::run(const std::string& token) {
     try {
         uint32_t intents = intents::i_default_intents | intents::i_guild_members;
         m_bot = std::make_unique<cluster>(token, intents);
+        m_bot->cache_policy = { cache_policy_setting_t::cp_lazy, cache_policy_setting_t::cp_lazy, cache_policy_setting_t::cp_aggressive };
         Handler::initialize(*m_bot.get());
-        m_bot->start(false);
+        m_bot->start(st_return);
+        m_is_connected = true;
     }
     catch (std::exception& e) {
         Handler::log_dpp("DPP thew an exception: " + (std::string)e.what(), "run()", ll_critical);
+        m_bot.reset();
```

Comment:
This is also dangerous. Though I think it can be fixed by covering the entire function with the lock.